sdk/nodejs/tests/provider/experimental/testdata/map-types/index.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright 2025-2025, Pulumi Corporation.  All rights reserved.
+
+import * as pulumi from "@pulumi/pulumi";
+
+export interface MyComponentArgs {
+    aRecordOfStrings: Record<string, string>;
+    aRecordOfNumbers?: Record<string, number>;
+    aRecordOfBooleans: Record<string, boolean>;
```

Comment:
Do we need/want to support Record? Do we ever emit it in codegen? What are its benefits over map?

Reply:
I think we want to support most types users want to use rather than what we emit in codegen.  I mainly added support here because https://github.com/pulumi/pulumi/pull/18546 and related issues, so I tried to think of all the map like types in typescript that we should support.  Not sure if there's any benefits or downsides over maps, but I don't think it hurts to err on the side of supporting more things here.

Reply:
Maybe? I think users will look at us to provide patterns, we won't support everything that they can think of, e.g. custom types that behave like maps or duck-type to maps. Supporting extra cases has cost for us to maintain, so I'd say if Map vs. Record is purely a taste choice, I would default to minimizing the options. We can also add them later if we find we need them.

Reply:
Honestly I don't know typescript well enough to understand the intricasies of both.  I'm fine with removing Record's for now, and then we can re-add them when users ask for it.